main/views.py

```diff
@@ -1,57 +1,137 @@
+import json
+import datetime
+
+
 from django.http import HttpResponse, JsonResponse
 from django.shortcuts import get_object_or_404, render
-
 from .models import *
 
 
 def index(request):
+    # i will remove matches and use results instead
     matches = Match.objects.all()
-    results = {}
+    results = []
     for match in matches:
         if match.player1_score > match.player2_score:
             winner = match.player1.name
             loser = match.player2.name
-            w = match.player1_score
-            l = match.player2_score
+            w_score = match.player1_score
```

Comment:
1. `w_score` still has an unexplained `w` in it - I recommend that you go ahead and type out `winner_score`.
2. Or not even save that into a variable. Since you already have the winner player referenced as `winner`, in your `results.append()` call you could just say `winner.score` and then you don't have to worry about whether that happens to be `player1` or `player2`.